lib/rules/no-unused-vars.js

```diff
@@ -155,7 +162,16 @@ module.exports = {
          * @returns {UnusedVarMessageData} The message data to be used with this unused variable.
          */
         function getAssignedMessageData(unusedVar) {
-            const additional = config.varsIgnorePattern ? `. Allowed unused vars must match ${config.varsIgnorePattern.toString()}` : "";
+            const def = unusedVar.defs && unusedVar.defs[0];
+            let patternToMatch;
+
+            if (config.destructuredArrayIgnorePattern && def.name.parent.type === "ArrayPattern") {
+                patternToMatch = config.destructuredArrayIgnorePattern;
+            } else if (config.varsIgnorePattern) {
+                patternToMatch = config.varsIgnorePattern;
+            }
+
+            const additional = patternToMatch ? `. Allowed unused vars must match ${patternToMatch.toString()}` : "";
```

Comment:
Should we provide a different message when `destructuredArrayIgnorePattern` is suggested, like "Allowed unused elements of array destructuring patterns must match..."?

Reply:
Yes, that will be a good enhancement.